TheMovieDbTests/MovieDBAPITests.swift

```diff
@@ -0,0 +1,198 @@
+import XCTest
+@testable import TheMovieDb
+
+class MovieDBAPITests: XCTestCase {
+    
+    var sut: MovieDBAPI!
+    
+    var networkDispatcher: MockNetworkDispatcher!
+
+    override func setUp() {
+        super.setUp()
+        networkDispatcher = MockNetworkDispatcher()
+        sut = MovieDBAPI(dispatcher: networkDispatcher)
+    }
+    
+    override func tearDown() {
+        sut = nil
+        networkDispatcher = nil
+        super.tearDown()
+    }
+    
+    func testSuccessExecuteWithCorrectData() {
+        networkDispatcher.data = """
+            {
+                "test_int": 0
+            }
+        """.data(using: .utf8)
+        
+        let expectation = self.expectation(description: "successExecute")
+        var result: Int?
+        sut.execute(SimpleRequest()) { response in
+            switch response {
+            case .success(let response):
+                result = response.testInt
+            default:
+                break
+            }
+            expectation.fulfill()
+        }
+        
+        waitForExpectations(timeout: 1, handler: nil)
+        XCTAssertEqual(result, 0)
+    }
+    
+    func testSuccessExecuteWithIncorrectData() {
+        networkDispatcher.data = """
+            {
+                "testint": 0
+            }
+        """.data(using: .utf8)
+        
+        let expectation = self.expectation(description: "successExecute")
+        var result: Error?
+        sut.execute(SimpleRequest()) { response in
+            switch response {
+            case .failure(let error):
+                result = error
+            default:
+                break
+            }
+            expectation.fulfill()
+        }
+        
+        waitForExpectations(timeout: 1, handler: nil)
+        XCTAssertNotNil(result as? NetworkError)
+    }
+    
+    func testFailureExecute() {
+        networkDispatcher.error = NetworkError.invalidRequest
+        
+        let expectation = self.expectation(description: "failureExecute")
+        var result: Error?
+        sut.execute(SimpleRequest()) { response in
+            switch response {
+            case .failure(let error):
+                result = error
+            default:
+                break
+            }
+            expectation.fulfill()
+        }
+        
+        waitForExpectations(timeout: 1, handler: nil)
+        XCTAssertNotNil(result as? NetworkError)
+    }
+    
+    func testGetRelatedMovies() {
+        networkDispatcher.data = """
+            {
+                "page": 1,
+                "results": [
+                  {
+                    "id": 299536,
+                    "overview": "As the Avengers...",
+                    "poster_path": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
+                    "release_date": "2018-04-25",
+                    "title": "Avengers: Infinity War",
+                  }
+                ],
+                "total_pages": 792,
+                "total_results": 15831
+              }
```

Comment:
In these cases you can make separate files and load them up in the test

Reply:
Ok